### PR TITLE
[TypeSpec] Remove "./" from tspconfig.yaml

### LINKED ISSUE #24196: [TypeSpec] Remove leading "./" from tspconfig.yaml values

### PR DESCRIPTION
**Pull request by mikeharder**
- Prefix is unnecessary and clutters config
- Fixes #24196 

Passing CI: https://dev.azure.com/azure-sdk/public/_build/results?buildId=2805265&view=results

### CODE REVIEW

**Comment by v-xuto** [comment]
> ### Generated ApiView
> Language	Package Name	ApiView Link
> Java	azure-ai-translation-text	Create ApiView failed. Please ensure your github account in [Azure](https://github.com/orgs/Azure/people)/[Microsoft](https://github.com/orgs/Microsoft/people) is public and add a comment "/azp run" to re-trigger the CI.
> Java	azure-contoso-widgetmanager	Create ApiView failed. Please ensure your github account in [Azure](https://github.com/orgs/Azure/people)/[Microsoft](https://github.com/orgs/Microsoft/people) is public and add a comment "/azp run" to re-trigger the CI.
> .Net	Azure.AI.Translation.Text	Create ApiView failed. Please ensure your github account in [Azure](https://github.com/orgs/Azure/people)/[Microsoft](https://github.com/orgs/Microsoft/people) is public and add a comment "/azp run" to re-trigger the CI.
> JavaScript	@azure-rest/ai-translation-text	Create ApiView failed. Please ensure your github account in [Azure](https://github.com/orgs/Azure/people)/[Microsoft](https://github.com/orgs/Microsoft/people) is public and add a comment "/azp run" to re-trigger the CI.
> JavaScript	@azure-rest/contoso-widgetmanager-rest	Create ApiView failed. Please ensure your github account in [Azure](https://github.com/orgs/Azure/people)/[Microsoft](https://github.com/orgs/Microsoft/people) is public and add a comment "/azp run" to re-trigger the CI.

@mikeharder The ApiView link generation failed, please ensure your github account in [Azure](https://github.com/orgs/Azure/people)/[Microsoft](https://github.com/orgs/Microsoft/people) is public.

File: specification/cognitiveservices/AnomalyDetector/tspconfig.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ emit: [
 ]
 options:
   "@azure-tools/typespec-autorest":
-    azure-resource-provider-folder: "./data-plane"
+    azure-resource-provider-folder: "data-plane"
     examples-directory: ./examples
```
Comment (weshaggard) [inline specification/cognitiveservices/AnomalyDetector/tspconfig.yaml:18]: one more case here to remove `./`

**Review by weshaggard — APPROVED** [review]

**Comment by v-xuto** [comment]
@qiaozha Please help review `azure-sdk-for-js` error.
@raych1  Please help review `azure-sdk-for-go` error.
@msyyc Please help review `azure-sdk-for-python` error.
@live1206 Please help review `azure-sdk-for-net-track2` error.

**Comment by raych1** [comment]
@v-xuto , GO isn't enabled for typespec DPG. 

JS and python failed for same reason and isn't related to this PR change.
```
[automation_generate.sh] /mnt/vss/_work/1/s/azure-rest-api-specs/specification/cognitiveservices/AnomalyDetector/client.tsp:6:17 - error invalid-ref: Namespace AnomalyDetector.Univariate doesn't have member DetectUnivariateEntireSeries
08:49:00.522 cmdout 	[automation_generate.sh] > 6 | @@convenientAPI(AnomalyDetector.Univariate.DetectUnivariateEntireSeries, false)
08:49:00.522 cmdout 	[automation_generate.sh]     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
08:49:00.522 cmdout 	[automation_generate.sh] Found 1 error, 4 warnings.
08:49:00.538 cmdout 	[automation_generate.sh] Traceback (most recent call last):
08:49:00.538 cmdout 	[automation_generate.sh]   File "/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/runpy.py", line 196, in _run_module_as_main

```

**Review by v-xuto — APPROVED** [review]

**Pull request merged**